spec/lib/formatter_spec.rb

```diff
@@ -126,8 +126,8 @@ module CodeClimate::TestReporter
         end
 
         it 'should include the path prefix if set' do
-          expect(formatter.short_filename('file1')).to eq('custom/file1')
-          expect(formatter.short_filename("#{::SimpleCov.root}/file1")).to eq('custom/file1')
+          expect(formatter.send(:short_filename, 'file1')).to eq('custom/file1')
+          expect(formatter.send(:short_filename, "#{::SimpleCov.root}/file1")).to eq('custom/file1')
```

Comment:
It's also an anti-pattern to have methods public that are not public :)
Atm it's 'incorrect' code and 'incorrect' tests and changing it to
'correct' code and 'incorrect' tests is somehow bad ?

changed it back to be 'incorrect' code, since this was supposed to be a
simple fix and not re-architecting :)


On Sat, Apr 18, 2015 at 6:12 PM, Noah Davis <notifications@github.com>
wrote:

> In spec/lib/formatter_spec.rb
> <https://github.com/codeclimate/ruby-test-reporter/pull/84#discussion_r28648889>
> :
>
> > @@ -126,8 +126,8 @@ module CodeClimate::TestReporter
> >          end
> >
> >          it 'should include the path prefix if set' do
> > -          expect(formatter.short_filename('file1')).to eq('custom/file1')
> > -          expect(formatter.short_filename("#{::SimpleCov.root}/file1")).to eq('custom/file1')
> > +          expect(formatter.send(:short_filename, 'file1')).to eq('custom/file1')
> > +          expect(formatter.send(:short_filename, "#{::SimpleCov.root}/file1")).to eq('custom/file1')
>
> I didn't understand before why you needed to change this spec -- now I see
> it's because of the private keyword.
>
> Generally however, it's a bit of an anti-pattern to test private methods.
> Do you just want to leave the method public for now and leave the spec
> alone? It's either that, or refactor in order to test this as a public
> method on a different object.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/codeclimate/ruby-test-reporter/pull/84/files#r28648889>
> .
>